library/core/src/num/int_macros.rs

```diff
@@ -2000,5 +2000,176 @@ macro_rules! int_impl {
         pub const fn max_value() -> Self {
             Self::MAX
         }
+
+        /// Adds to the current value, returning the previous value.
+        ///
+        /// This operation wraps around on overflow.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_fetch_ops)]
+        #[doc = concat!("let mut foo: ", stringify!($SelfT), " = 0;")]
+        /// assert_eq!(foo.fetch_add(10), 0);
+        /// assert_eq!(foo, 10);
+        /// ```
+        #[inline]
+        #[unstable(feature = "int_fetch_ops", issue = "none")]
+        pub fn fetch_add(&mut self, val: Self) -> Self {
+            let old = *self;
+            *self = self.wrapping_add(val);
+            old
+        }
+
+        /// Subtracts from the current value, returning the previous value.
+        ///
+        /// This operation wraps around on overflow.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_fetch_ops)]
+        #[doc = concat!("let mut foo: ", stringify!($SelfT), " = 20;")]
+        /// assert_eq!(foo.fetch_sub(10), 20);
+        /// assert_eq!(foo, 10);
+        /// ```
```

Comment:
Maybe a warning at the end of each doc comment that this is not an atomic operation (the name being the same as the atomic one, I could see it confusing people) ?

I don't think there is a precedent for that though, and static+strict typing make it less likely to be a problem in Rust, so maybe it's overkill

Reply:
So I initially agreed with this idea, but I've realised I'm increasingly reluctant to do it. Going from simple integer arithmetic to notions of thread safety and atomicity is a bit of a step up, and the `&mut` bound is already preventing use of this function in such contexts. I'm struggling to introduce this information in a manner that doesn't sound out of place or like there's some inherent risk when using these functions.

Reply:
Overkill it is then ! I just wanted to put the idea out, but I'm not fixated on it, and you're right that &mut does most of the work already